cli/init.js

```diff
@@ -18,11 +18,18 @@ module.exports = function init(args) {
   const relativePublicDir = path.relative(CWD, absolutePublicDir)
   const dirExists = fs.existsSync(absolutePublicDir)
 
-  invariant(
-    dirExists,
-    'Failed to create a Service Worker at "%s": directory does not exist.\nMake sure to include a relative path to the root directory of your server.',
-    absolutePublicDir,
-  )
+  if (!dirExists) {
+    // Try to create the directory if it doesn't exist
+    try {
+      fs.mkdirSync(absolutePublicDir, { recursive: true })
```

Comment:
I suggest we use its Promise API and wrap it in `until` for better error handling.

```js
const { until } = require('@open-draft/until')

module.exports = async function init(args) {
  const [createDirectoryError] = await until(() => fs.promises.mkdir(...))

  invariant(createDirectoryError, '...')
}
```